### ISSUE TITLE
Making DetectNet sensitive to objects smaller than 50x50 pixels

### ISSUE THREAD

**Issue by DarylWM** [NVIDIA/DIGITS#1115]
The objects I'm aiming to detect are much smaller than the 50x50 pixels DetectNet detects. The smallest object in my training set is 9x4 pixels. I've had success by magnifying the images so the smallest objects are at least 50x50, but how would I modify DetectNet to detect smaller objects?

**Comment by ontheway16** [comment]
I am on the same boat. I am planning to detect objects in a large set of scanned images (background color is more or less solid (single-colored) and different then objects' color) and It appears my object dimensions are between 20 to 60 pixels, so they are smaller then whats advised (50 to 400 pixels). Since there are over 2000 scanned images, before starting the labelling, I am looking for the best way on how to optimize input images for higher accuracy and lesser process duration..
One option is upscaling the original images in an image processing software, to make object dimensions double, like bringing them around 40x40 to 120x120. But this is not a good approach due to loss of image quality, especially for smaller objects.
Other option is to take bounding box bigger, say, if the object dimension is 23x41 pixels, I can find the approximate center of object and draw a 50x50 bounding box around it, and avoid labeling the objects staying too close to each other, in the train set.
Another question is, size of training images. my current scan sizes are around 5000x10000 pixels. Is it possible to feed these to DetectNet as is, or should I create several cropped images from each image (in an image processing software) then start labeling? If cropping necessary, which size may be the best? 1024x1204? 1248x384?
I appreciate any guidance since labeling work will be very time consuming and do not want to make a mistake which eventually render labelling efforts useless.

**Comment by gheinrich** [comment]
To be safe, if I were you I would make a note of the actual contours of objects during the annotation process. This way, you are not tied to one particular object detection scheme. For DetectNet it would be pretty straightforward to figure out the coordinates of bounding boxes from your contour annotations. You might also want to approach your problem as an image segmentation task. Image segmentation should work pretty much out of the box if there is a clear delineation between objects in your images. You would then "just" have to do the clustering and post processing yourself.

**Comment by ontheway16** [comment]
@gheinrich Thanks for the answer, since objects their selves and background more or less clearly seperated, I can easily apply particle analysis tools to reveal topleft corners of particles (which are my objects in this case), and, width and height of them so I believe it will be easier to generate (and capture the coords of) a tight bounding box around them.
Here is a sample close-up of target objects;

https://www.dropbox.com/s/r83ut3ge9v8fnoh/IMG_20160617_0113-2.tif?dl=0

I am aiming to differentiate the blacktailed ones (there are three) from others.

With image segmentation, do you mean to follow a route other then deep learning? Thats also possible, but several preprocessing steps involves to make it, for each and every image, which costs a lot of processing time for each scan, and results are not quite reliable. If this works as planned, It will be possible to feed unretouched (or just with a small retouch work) images, directly to this system and it seems detection speed is faster.

**Comment by gheinrich** [comment]
> With image segmentation, do you mean to follow a route other then deep learning?

I was suggesting to use an image segmentation neural net - no pre-processing involved. Examples:
https://github.com/NVIDIA/DIGITS/tree/digits-5.0/examples/binary-segmentation
https://github.com/NVIDIA/DIGITS/tree/digits-5.0/examples/semantic-segmentation
https://github.com/NVIDIA/DIGITS/tree/digits-5.0/examples/medical-imaging

**Comment by ontheway16** [comment]
@gheinrich Well, many thanks for the guidance. What I understand is, semantic-segmentation appears the most useful one for me. Since I am not a programmer, I have to find most useful solution among whats currently available. Do you know if semantic-segmentation code is able to create a text-list of objects (classes) detected, along with information on how many objects detected from each class, for a given image? (ie, car 5, sheep 2, people 4,.. etc.)

**Comment by gheinrich** [comment]
The output of a segmentation network is an image that is as big as the input image. Each pixel in the output image denotes the class of the corresponding pixel in the input image. If you are not a programmer you may find it easier to use DetectNet then, which will produce a list of objects.

**Comment by ontheway16** [comment]
@gheinrich again thanks for the advises. I could not find enough information to decide for segmentation network and already decided to follow only remaining route, directnet. Since I am not very much into these (yet), while detectnet sample continues its training in computer, I am nowadays trying to figure out what actually the detectnet is..
Is it just a model that runs under Digits?
Or, a model + some other pieces of software tailored for a specific job?
Is it build just to show abilities of Digits while allowing us to manipulate some parameters for experimenting with our own datasets, or is it created to become a basis for object recognition  for Digits, with ongoing development.....

These are the questions in mind. I spend for a good amount for a Titan X for this purpose without  knowing how things will develop for non-programmers...
